crates/router/src/connector/iatapay/transformers.rs

```diff
@@ -97,27 +143,30 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for IatapayPaymentsRequest {
             | PaymentMethod::BankDebit
             | PaymentMethod::Reward
             | PaymentMethod::Voucher
-            | PaymentMethod::GiftCard => item.get_billing_country()?.to_string(),
+            | PaymentMethod::GiftCard => item.router_data.get_billing_country()?.to_string(),
         };
-        let return_url = item.get_return_url()?;
-        let payer_info = match item.request.payment_method_data.clone() {
+        let return_url = item.router_data.get_return_url()?;
+        let payer_info = match item.router_data.request.payment_method_data.clone() {
             api::PaymentMethodData::Upi(upi_data) => upi_data.vpa_id.map(|id| PayerInfo {
                 token_id: id.switch_strategy(),
             }),
             _ => None,
         };
-        let amount =
-            utils::to_currency_base_unit_asf64(item.request.amount, item.request.currency)?;
+        let amount = utils::to_currency_base_unit_asf64(
+            item.router_data.request.amount,
+            item.router_data.request.currency,
+        )?;
```

Comment:
Since we have implemented the transformation now you can directly use `item.amount`
No need of this `utils::to_currency_base_unit_asf6` call, and in hindsight this is the main task of this PR